### PR TITLE
Refactor Localdate by replacing dateTime with eventDateTime and move …

### PR DESCRIPTION
**Pull request by alinaitschool**
**This fix contain:**

- add @JsonIgnore for fixing NullPointerException
- create new packet for exception and move EventNotFoundException
- replace 500 status code with 404 for user_id = Null
- refactor dateTime by changing into eventDateTime and delete dateTime since were 2 attribute in event class for LocalDateTime 
- fix repositories query by refactoring from dateTime to EventDateTime

### CODE REVIEW

**Comment by alinaitschool** [comment]
@AlexandruIliescu please review this PR. Thanks

File: src/main/java/com/itschool/eventmanagment/models/entities/Event.java [hunk 1]
```diff
@@ -25,12 +26,10 @@ public class Event {
     private Long id;
     @Column
     private String name;
-    @Column (name = "event_date_time")
-    private LocalDateTime eventDateTime;
     @JsonDeserialize(using = LocalDateTimeDeserializer.class)
     @JsonSerialize(using = LocalDateTimeSerializer.class)
```
Comment (AlexandruIliescu) [inline src/main/java/com/itschool/eventmanagment/models/entities/Event.java:30]: Please consider to remove `@Json...` annotation from the entity. This class is used to represent the database structure, so is not directly interacting with JSON serialization/deserialization. We should keep those annotations only on DTOs.

File: src/main/java/com/itschool/eventmanagment/services/EventServiceImp.java [hunk 2]
```diff
@@ -59,9 +59,9 @@ public List<EventDTO> getEvents() {
     public List<EventDTO> getEvents(LocalDateTime to, LocalDateTime from, String location, String sortBy) {
         List<Event> events;
         if ("location".equalsIgnoreCase(sortBy)) {
-            events = eventRepository.findByDateTimeBetweenAndLocationContainingIgnoreCaseOrderByLocationAsc(to, from, location);
+            events = eventRepository.findByEventDateTimeBetweenAndLocationContainingIgnoreCaseOrderByLocationAsc(to, from, location);
         } else {
-            events = eventRepository.findByDateTimeBetweenAndLocationContainingIgnoreCaseOrderByDateTime(to, from, location);
+            events = eventRepository.findByEventDateTimeBetweenAndLocationContainingIgnoreCaseOrderByEventDateTime(to, from, location);
         }
         return events.stream().map(event -> objectMapper.convertValue(event, EventDTO.class)).collect(Collectors.toList());
```
Comment (AlexandruIliescu) [inline src/main/java/com/itschool/eventmanagment/services/EventServiceImp.java:66]: Can we maybe simplify a bit this stream using just `toList()` method?

```suggestion
        return events.stream().map(event -> objectMapper.convertValue(event, EventDTO.class)).toList();
```

File: src/main/java/com/itschool/eventmanagment/models/dtos/EventDTO.java [hunk 3]
```diff
@@ -13,12 +13,11 @@
 public class EventDTO {
     private Long id;
     private String name;
-    //private LocalDateTime eventDateTime;
     private String location;
     @JsonDeserialize(using = LocalDateTimeDeserializer.class)
     @JsonSerialize(using = LocalDateTimeSerializer.class)
     private LocalDateTime eventDateTime;
     private Long userId;
```
Comment (AlexandruIliescu) [inline src/main/java/com/itschool/eventmanagment/models/dtos/EventDTO.java:20]: If we have a property called `maxParticipants`, should we also have `participantId`? Maybe we can try to merge somehow the naming of these two properties so we can have a better intent with the properties.

User in this DTO doesn't say much. User of what? User of an event? An event can have participants, I guess. Those are some questions that I asked myself when I saw this `userId` property.

Also, this id property can be among the first properties that we have; we can consider having it under the first property of this class (id).

File: src/main/java/com/itschool/eventmanagment/models/entities/Event.java [hunk 4]
```diff
@@ -25,12 +26,10 @@ public class Event {
...
-    @Column
-    private LocalDateTime dateTime;
+    @Column (name = "event_date_time")
+    private LocalDateTime eventDateTime;
     @Column
     private String location;
     @Column (name = "max_participants")
```
Comment (AlexandruIliescu) [inline src/main/java/com/itschool/eventmanagment/models/entities/Event.java:35]: It's not strictly necessary to include business logic methods like `registerParticipant()` in the entity itself. In many cases, you might move this kind of logic to a service class (e.g., EventServiceImpl) to follow the separation of concerns principle, which suggests that entities should mainly represent data and relationships, while business logic should be handled by services.

I could not comment on `registerParticipant()` method from this class, from line 50, so I made a coment on the property.

**Review by AlexandruIliescu — APPROVED** [review]

File: src/main/java/com/itschool/eventmanagment/models/entities/Event.java (hunk 1 above)
Comment (alinaitschool) [reply src/main/java/com/itschool/eventmanagment/models/entities/Event.java:30]: If I removing I will have a 500 error.

Screenshot 2024-09-19 at 12 07 39
Screenshot 2024-09-19 at 12 07 47

File: src/main/java/com/itschool/eventmanagment/models/dtos/EventDTO.java (hunk 3 above)
Comment (alinaitschool) [reply src/main/java/com/itschool/eventmanagment/models/dtos/EventDTO.java:20]: @AlexandruIliescu for the first point I didn't get it, but I can keep this in mind and  ask
"If we have a property called maxParticipants, should we also have participantId? Maybe we can try to merge somehow the naming of these two properties so we can have a better intent with the properties."

This part with UserDTO, also is a bit confusing that was in the specification to use a user which create a event. 

Regarding the id it's was add it by Denisa but I've move in the top of the list.

**Pull request merged**